### PR TITLE
feat(suite-web): use data.trezor.io for FW hash check

### PR DESCRIPTION
**Pull request by Lemonexe**
## Description

Currently suite-web downloads FW binary for FW hash check from suite.trezor.io, where only some binaries are available, and it cannot be relied upon.

:arrow_right:  switch to data.trezor.io where all binaries are available so all [supported versions](https://github.com/trezor/trezor-suite/blob/0fd023dbe89e77037f7ab75118ee10d83047449b/packages/connect/src/constants/firmware.ts) can be checked

## Related Issue

Resolve https://github.com/trezor/trezor-suite-private/issues/122

## QA instructions

- visit https://dev.suite.sldev.cz/suite-web/feat/fw-hash-check-from-data-on-web/web/settings/debug
- Turn **OFF revision** check
- Turn **ON hash** check
- Open `trezor-user-env`, try with following devices. The expected results are that it should either pass :heavy_check_mark: , or get ghost modal :ghost: with "Firmware hash check failed. Your Trezor might be counterfeit." banner after clicking Back
  - T1B1
    - 1.10.5 :heavy_check_mark: 
    - 1.11.1 :ghost: 
    - 1.12.1 :ghost: 
  - T2T1
    - 2.3.0 :heavy_check_mark: 
    - 2.5.1 :ghost: 
    - 2.5.2 :ghost: 
    - 2.8.1 :ghost: 
  - T2B1 
    - 2.8.0 :ghost: 
  - T3T1 
    - 2.8.1 :ghost:

### CODE REVIEW

File: packages/connect/src/api/firmware/getBinary.ts [hunk 1]
```diff
@@ -9,7 +9,7 @@ interface GetBinaryProps {
 
 export const getBinary = ({ baseUrl, btcOnly, release }: GetBinaryProps) => {
     const fwUrl = release[btcOnly ? 'url_bitcoinonly' : 'url'];
-    const url = `${baseUrl}/${fwUrl}`;
+    const url = `${baseUrl.replace(/\/+$/, '')}/${fwUrl}`;
```
Comment (peter-sanderson) [inline packages/connect/src/api/firmware/getBinary.ts]: Just idea to move this `/\/+$/` to named constant to explain it without need to decipher the regex =>

```ts
const ALL_SLASHES_AT_THE_END_REGEX = /\/+$/
```

**Review by peter-sanderson — APPROVED** [review]

File: packages/connect/src/api/firmware/getBinary.ts (hunk 1 above)
Comment (Lemonexe) [reply packages/connect/src/api/firmware/getBinary.ts]: https://github.com/trezor/trezor-suite/pull/15508/commits/65dba8dbdee07628643e81e76e0ec88cfdccb794

**Review by mroz22 — APPROVED** [review]
ok did not test

**Comment by Lemonexe** [comment]
Putting on hold, TBD if we want this. EDIT: yes we want this :+1: 

Also it was not ready for merge, because these tests are failing:
```
t3t1-create-addtional-share.test.ts
t2t1-recovery-persistence.test.ts
t2t1-recovery-success.test.ts
```

File: packages/connect/src/api/firmware/getBinary.ts [hunk 2]
```diff
@@ -1,6 +1,8 @@
 import { httpRequest } from '../../utils/assets';
 import { FirmwareRelease } from '../../types';
 
+const ALL_SLASHES_AT_THE_END_REGEX = /\/+$/;
```
Comment (peter-sanderson) [inline packages/connect/src/api/firmware/getBinary.ts:4]: :clap:

**Review by peter-sanderson — APPROVED** [review]

**Pull request merged**